File: receipt-analyzer-service/src/main/java/pl/madzierski/daniel/app/receipt/revision/model/UpdateRevisionRequest.java
package pl.madzierski.daniel.app.receipt.revision.model;

import pl.madzierski.daniel.app.receipt.scan_resolver.ReceiptResolverStrategyType;

import java.time.LocalDateTime;
import java.util.Set;

public record UpdateRevisionRequest(
        String id,
        ReceiptResolverStrategyType resolver,
        LocalDateTime createdDate,
        String brand,
        Double totalPrice,
        String payingDate,
        String address,
        Boolean isPreferredRevision,
        Boolean isCorrect,
        Set<ItemResponse> items
) {

    public record ItemResponse(
            String id,
            String name,
            Double amount,
            Double unitPrice,
            Double discount,
            Double totalPrice,
            Integer position
    ) {
    }
}

File: receipt-analyzer-service/src/main/kotlin/pl/madzierski/daniel/app/receipt/revision/model/UpdateRevisionRequest.kt
package pl.madzierski.daniel.app.receipt.revision.model

import pl.madzierski.daniel.app.receipt.scan_resolver.ReceiptResolverStrategyType
import java.time.LocalDateTime

data class UpdateRevisionRequest(
    var id: String?,
    var resolver: ReceiptResolverStrategyType?,
    val createdDate: LocalDateTime?,
    val brand: String?,
    val totalPrice: Double?,
    val payingDate: String?,
    val address: String?,
    val isPreferredRevision: Boolean?,
    val isCorrect: Boolean?,
    val items: Set<ItemResponse>?,
) {

    data class ItemResponse(
        val id: String?,
        val name: String?,
        val amount: Double?,
        val unitPrice: Double?,
        val discount: Double?,
        val totalPrice: Double?,
        val position: Int?,
    )

}
